src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -429,6 +434,80 @@ def generate_request_payload(configuration, location, public_key, tags):
     return cc
 
 
+def get_kubeconfig_dict(kube_config=None):
+    # Gets the kubeconfig as per kubectl(after applying all merging rules)
+    args = ['kubectl', 'config', 'view']
+    if kube_config:
+        args += ["--kubeconfig", kube_config]
+
+    # subprocess run
+    try:
+        proc = run(args, stdout=PIPE, stderr=STDOUT, universal_newlines=True)
```

Comment:
Can we find a way to figure out if its an aks cluster which dosent involve using subprocess and use python SDK instead?

Reply:
Okay, will have a look again.

Reply:
Removed subprocess run approach. Please have a look at it.